### ISSUE TITLE
Recommend replace off the resize that is not corner aligned

### ISSUE THREAD

**Issue by XiaLiPKU** [dmlc/gluon-cv#437]
https://github.com/dmlc/gluon-cv/blob/d298296d721abc8257e226bfc7303d24cdd32145/gluoncv/data/segbase.py#L61

As I have tested, bilinear resize operations in opencv3.3.1 and PIL.Image are not corner aligned.
Meanwhile, the BilinearResize2D in gluon is corner aligned.

They are contradictory for semantic segmentation:
In training, the images are downsample without corner alignment.
While in testing, the predicted labels are upsampled with alignment.
It's clear to see, these operations are not symmetric.

In tensorflow' officail deeplab implementation(as following), both downsample for training and upsampling of predicted labels are corner aligned, and I think it is a better choice.

https://github.com/tensorflow/models/blob/f175abc3457b6c8aec01a8b5c53adc8de5a4111d/research/deeplab/input_preprocess.py#L87
https://github.com/tensorflow/models/blob/f175abc3457b6c8aec01a8b5c53adc8de5a4111d/research/deeplab/model.py#L187

**Comment by zhanghang1989** [comment]
As far as I know, `BilinearResize2D` does not align the corner.

**Comment by XiaLiPKU** [comment]
https://github.com/dmlc/gluon-cv/issues/437#issuecomment-433195125
> As far as I know, `BilinearResize2D` does not align the corner.

To prove my statement, I run some scripts as below.
Here, the versions of corrlated libraries in my server are:
python-3.6.4
ipython-6.4.0
torch-1.0.0.dev20181005
mxnet-1.3.0
cv2-3.1.0
PIL-5.2.0


![Image of Yaktocat](https://raw.githubusercontent.com/XiaLiPKU/Dive-into-Bilinear/master/pytorch_oup.png)
![Image of Yaktocat](https://raw.githubusercontent.com/XiaLiPKU/Dive-into-Bilinear/master/gluon_oup.png)
![Image of Yaktocat](https://raw.githubusercontent.com/XiaLiPKU/Dive-into-Bilinear/master/cv2_oup.png)
![Image of Yaktocat](https://raw.githubusercontent.com/XiaLiPKU/Dive-into-Bilinear/master/PIL_oup.png)

It's clear that BilinearResize2D align the corners

**Comment by XiaLiPKU** [comment]
> As far as I know, `BilinearResize2D` does not align the corner.

Moreover, I see in your repo gluoncv-torch, the defauly upsample setting for align_corners is True.
https://github.com/tensorflow/models/blob/f175abc3457b6c8aec01a8b5c53adc8de5a4111d/research/deeplab/model.py#L187

As in my own implementation of segmentor (single Deeplabv2 without coco pretrained), upsample using aligned version can lead to around 1 point (72->72) in mIoU compared to not aligned one. So, of course, align_corners is the right choice. But I just wonder should the downsample for training image also be corner aligned.

One more thing to say, the default Interp layer in Caffe recommands the height and width of feature map to be odd numbers, like 321, 161, 81, 41 in Deeplabv2
https://bitbucket.org/aquariusjay/deeplab-public-ver2/src/master/src/caffe/layers/interp_layer.cpp#interp_layer.cpp-40.
Because align_corners doesn't propotionally upsample the image when sizes are even numbers, odd number if preferable.
So, I also recommend setting the default crop size from 480 to 481, default base size from 520 to 521, to be consistent with the corner aligned property of BilinearResize2D.

This suggestion also applies to your repo gluoncv-torch
